featurebyte/service/online_store_compute_query_service.py

```diff
@@ -0,0 +1,92 @@
+"""
+OnlineStoreComputeQueryService class
+"""
+from __future__ import annotations
+
+from typing import AsyncIterator
+
+from featurebyte.exception import DocumentNotFoundError
+from featurebyte.models.online_store_compute_query import OnlineStoreComputeQueryModel
+from featurebyte.schema.common.base import BaseDocumentServiceUpdateSchema
+from featurebyte.service.base_document import BaseDocumentService
+
+
+class OnlineStoreComputeQueryService(
+    BaseDocumentService[
+        OnlineStoreComputeQueryModel,
+        OnlineStoreComputeQueryModel,
+        BaseDocumentServiceUpdateSchema,
+    ]
+):
+    """
+    OnlineStoreComputeQueryService
+
+    The collection keeps track of currently active queries that are being run periodically to update
+    online store tables.
+    """
+
+    document_class = OnlineStoreComputeQueryModel
+
+    async def list_by_aggregation_id(
+        self, aggregation_id: str
+    ) -> AsyncIterator[OnlineStoreComputeQueryModel]:
+        """
+        List all documents by aggregation_id
+
+        Parameters
+        ----------
+        aggregation_id: str
+            Aggregation id
+
+        Yields
+        ------
+        list[OnlineStoreComputeQueryModel]
+            List of OnlineStoreComputeQueryModel
+        """
+        async for model in self.list_documents_iterator(
+            query_filter={"aggregation_id": aggregation_id}
+        ):
+            yield model
+
+    async def list_by_result_names(
+        self, result_names: list[str]
+    ) -> AsyncIterator[OnlineStoreComputeQueryModel]:
+        """
+        List all documents by aggregation result names
+
+        Parameters
+        ----------
+        result_names: list[str]
+            Result ids
```

Comment:
is result name the same as result IDs? might be worth making consistent here in the docstring, and below in `delete_by_result_name` too

Reply:
Yes they are the same. Will update the docstring to be consistent 👍